instance/tests/integration/integration_instance.py

```diff
@@ -56,3 +58,52 @@ def test_provision_instance(self):
         )
         provision_instance(instance.pk)
         self.assertEqual(instance.status, 'ready')
+
+
+class InstanceIntegrationFailureTestCase(IntegrationTestCase):
+    """
+    Make sure failures to the deployment process are handled
+    """
+    def test_ansible_failure(self):
+        """
+        Ensure failures in the ansible flow are reflected in the instance
+        """
+        with mock.patch('git.refs.head.Head.checkout'):
+            with mock.patch('git.Git.working_dir', new_callable=mock.PropertyMock) as wd:
+                wd.return_value = os.path.join(os.path.dirname(__file__), "ansible")
+
+                uid = str(uuid.uuid4())[:8]
+                instance = OpenEdXInstance.objects.create(
```

Comment:
Since this duplicates code from the other integration test, it could be worth factorizing it into a instance factory for integration tests?